### PR TITLE
LibWeb: Implement enough input events to send message on Discord

### PR DESCRIPTION
**Pull request by kalenikaliaksandr**
Implements:
- "input" and "beforeinput" events dispatch
- fix for finding correct cursor position when contenteditable receives a focus

### CODE REVIEW

File: Userland/Libraries/LibWeb/Page/EventHandler.cpp [hunk 1]
```diff
@@ -1086,6 +1096,18 @@ EventResult EventHandler::handle_keydown(UIEvents::KeyCode key, u32 modifiers, u
...
+            EventResult before_input_event_result;
+            if (key == UIEvents::KeyCode::Key_Return) {
+                before_input_event_result = fire_input_event(UIEvents::EventNames::beforeinput, UIEvents::InputTypes::insertParagraph, m_navigable, code_point);
+            } else {
+                before_input_event_result = fire_input_event(UIEvents::EventNames::beforeinput, UIEvents::InputTypes::insertText, m_navigable, code_point);
+            }
+            if (before_input_event_result == EventResult::Cancelled) {
+                return EventResult::Cancelled;
+            }
+        }
```
Comment (trflynn89) [inline Userland/Libraries/LibWeb/Page/EventHandler.cpp:1109]: Feels fragile having this here, if another key handler is inserted after the Key_Return handler below. Let's just bundle the `insertParagraph` event within the branch that handles return, and the `insertText` within the branch that handles text?

Comment (trflynn89) [reply Userland/Libraries/LibWeb/Page/EventHandler.cpp:1109]: As an aside, this method is getting a bit hard to follow...as a future cleanup, we might want to add a TRY-like macro that fires events and returns automatically if cancelled, so all of these are simple one-liners

(I'm envisioning something like):
```c++
FIRE(input_event(UIEvents::EventNames::beforeinput, UIEvents::InputTypes::insertParagraph, m_navigable, code_point));
```

File: Tests/LibWeb/Text/expected/contenteditable-should-insert-into-nested-node.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+hello!!!
```
[the remark this answers is not in the document]
Comment (ADKaster) [reply Tests/LibWeb/Text/expected/contenteditable-should-insert-into-nested-node.txt:1]: first message as well "conteneditable"

File: Userland/Libraries/LibWeb/Internals/Internals.cpp [hunk 3]
```diff
@@ -93,7 +93,7 @@ void Internals::send_key(HTML::HTMLElement& target, String const& key_name, WebI
 
 void Internals::commit_text()
 {
-    internals_page().handle_keydown(UIEvents::Key_Return, 0, 0);
+    internals_page().handle_keydown(UIEvents::Key_Return, 0, 0x0D);
```
[the remark this answers is not in the document]
Comment (kalenikaliaksandr) [reply Userland/Libraries/LibWeb/Internals/Internals.cpp]: I initially added it because `should_ignore_keydown_event()` makes a decision based on code point and it was returning true for "return" key, when it is send by `internals.sendText()`. I dropped this commit because it's no longer needed after I rearranged the code.

File: Userland/Libraries/LibWeb/Page/EventHandler.cpp (hunk 1 above)
Comment (kalenikaliaksandr) [reply Userland/Libraries/LibWeb/Page/EventHandler.cpp]: good idea, I added a FIRE macro.

File: Userland/Libraries/LibWeb/Page/EventHandler.cpp [hunk 4]
```diff
@@ -929,6 +929,8 @@ EventResult EventHandler::handle_keydown(UIEvents::KeyCode key, u32 modifiers, u
         dispatch_result = fire_keyboard_event(UIEvents::EventNames::keypress, m_navigable, key, modifiers, code_point);
         if (dispatch_result != EventResult::Accepted)
             return dispatch_result;
+
+        FIRE(input_event(UIEvents::EventNames::beforeinput, UIEvents::InputTypes::insertText, m_navigable, code_point));
```
[the remark this answers is not in the document]
Comment (kalenikaliaksandr) [reply Userland/Libraries/LibWeb/Page/EventHandler.cpp]: I just checked and this case works correctly, but I agree that having it closer to `handle_insert()` is good idea so moved it there.

**Review by trflynn89 — APPROVED** [review]
